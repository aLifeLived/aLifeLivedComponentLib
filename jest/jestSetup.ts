/* eslint-disable no-undef */
jest.useFakeTimers({ legacyFakeTimers: true }); // Stops reference error from appearing
