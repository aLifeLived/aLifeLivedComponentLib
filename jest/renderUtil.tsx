import React from 'react';
import { ThemeProvider } from '../src/themes/ThemeProvider';
import { render as rtlRender } from '@testing-library/react-native';
import { library } from '@fortawesome/fontawesome-svg-core';
import { iconList } from '../src/themes/systemIcon/IconList';

library.add(iconList);

function render(ui: any, { ...renderOptions } = {}) {
  function Wrapper({ children }: { children: any }) {
    return <ThemeProvider>{children}</ThemeProvider>;
  }

  return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}

// re-export everything for convenience
export * from '@testing-library/react-native';

// override render method
export { render };
