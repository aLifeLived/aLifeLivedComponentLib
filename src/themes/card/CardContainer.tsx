import React, { ReactNode } from 'react';

// UI
import { Box } from '../box/Box';
import { ColorThemeTokens } from '../palette';
import { SystemButton } from '../systemButton/SystemButton';

type CardContainerProps = {
  onPress: () => void;
  backgroundColor?: ColorThemeTokens;
  accessibilityLabel?: string;
  testID?: string;
  children: ReactNode;
};

export const CardContainer: React.FC<CardContainerProps> = ({
  onPress,
  backgroundColor = 'backgroundXXLow',
  accessibilityLabel,
  testID,
  children,
}) => {
  return (
    <SystemButton
      onPress={onPress}
      alignContent="space-between"
      borderRadius="large"
      backgroundColor={backgroundColor}
      accessibilityLabel={accessibilityLabel}
      testID={testID}
    >
      <Box>{children}</Box>
    </SystemButton>
  );
};
