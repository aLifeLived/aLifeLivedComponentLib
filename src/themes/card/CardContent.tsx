import React, { ReactNode } from 'react';

// UI
import { Box } from '../box/Box';

type CardContentProps = {
  SVG: ReactNode;
};

export const CardContent: React.FC<CardContentProps> = ({ SVG }) => {
  return (
    <Box alignContent="center" overflow="hidden" borderRadius="large">
      {SVG}
    </Box>
  );
};
