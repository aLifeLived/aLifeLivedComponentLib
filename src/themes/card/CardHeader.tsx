import React from 'react';

// UI
import { Box } from '../box/Box';
import { Text } from '../text/Text';

type CardHeaderProps = {
  title: string;
};

export const CardHeader: React.FC<CardHeaderProps> = ({ title }) => {
  return (
    <Box paddingRight="m" paddingTop="s" alignSelf="flex-end">
      <Text variant="h4">{title}</Text>
    </Box>
  );
};
