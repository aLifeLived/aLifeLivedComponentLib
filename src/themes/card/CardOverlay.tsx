import React from 'react';

// UI
import { BlurLayout } from '../systemBlurLayout/SystemBlurLayout';
import { Text } from '../text/Text';

type CardOverlayProps = {
  title: string;
};

export const CardOverlay: React.FC<CardOverlayProps> = ({ title }) => {
  return (
    <BlurLayout
      tint="dark"
      intensity={80}
      backgroundColor="blue"
      position="absolute"
      bottom={0}
      left={0}
      right={0}
      padding="s"
      borderRadius="large"
      overflow="hidden"
    >
      <Text textAlign="center" color="textContrastXXLow" variant="h4">
        {title}
      </Text>
    </BlurLayout>
  );
};
