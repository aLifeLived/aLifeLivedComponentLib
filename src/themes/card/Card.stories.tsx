import React from 'react';
import { storiesOf } from '@storybook/react-native';

// SVG
import MyStoriesSVG from '../../assets/svg/my-stories.svg';
import DraftsSVG from '../../assets/svg/drafts.svg';
import TravelThemeSVG from '../../assets/svg/travel-theme.svg';
import COVIDThemeSVG from '../../assets/svg/covid-theme.svg';
import LifeStorySVG from '../../assets/svg/life-story-theme.svg';

// Component
import { CardContainer } from './CardContainer';
import { CardHeader } from './CardHeader';
import { CardContent } from './CardContent';
import { CardOverlay } from './CardOverlay';

// UI
import { Box } from '../box/Box';
import { SystemScrollView } from '../systemScrollView/SystemScrollView';

storiesOf('Cards', module)
  .addDecorator(Story => Story())
  .add('Card with headers', () => (
    <Box flex={1}>
      <Box
        justifyContent="center"
        alignContent="center"
        alignItems="center"
        flexDirection="row"
        flexWrap="wrap"
      >
        <Box marginRight="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="primaryOverlay"
          >
            <CardHeader title="My stories" />
            <CardContent SVG={<MyStoriesSVG height={100} />} />
          </CardContainer>
        </Box>

        <Box marginBottom="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardHeader title="Travel" />
            <CardContent SVG={<TravelThemeSVG height={100} />} />
          </CardContainer>
        </Box>

        <Box marginRight="m" width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="COVID" />
            <CardContent SVG={<COVIDThemeSVG />} />
          </CardContainer>
        </Box>

        <Box width="40%">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="Drafts" />
            <CardContent SVG={<DraftsSVG height={90} />} />
          </CardContainer>
        </Box>

        <Box width="40%" marginTop="m">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="backgroundXXLow"
          >
            <CardHeader title="Lift Story" />
            <CardContent SVG={<LifeStorySVG height={90} />} />
          </CardContainer>
        </Box>
      </Box>
    </Box>
  ))
  .add('Card with overlay text', () => (
    <SystemScrollView alignContent="center" flex={1}>
      <Box
        flexDirection="row"
        marginLeft="m"
        marginRight="m"
        justifyContent="center"
        flexWrap="wrap"
      >
        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<MyStoriesSVG height={150} width={150} />} />
            <CardOverlay title="School years" />
          </CardContainer>
        </Box>
        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<DraftsSVG height={150} width={150} />} />
            <CardOverlay title="Early life" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<TravelThemeSVG height={150} width={150} />} />
            <CardOverlay title="School years" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<COVIDThemeSVG height={150} width={150} />} />
            <CardOverlay title="Sports" />
          </CardContainer>
        </Box>

        <Box margin="s">
          <CardContainer
            onPress={() => console.log('press')}
            backgroundColor="secondaryOverlay"
          >
            <CardContent SVG={<LifeStorySVG height={150} width={150} />} />
          </CardContainer>
        </Box>
      </Box>
    </SystemScrollView>
  ));
