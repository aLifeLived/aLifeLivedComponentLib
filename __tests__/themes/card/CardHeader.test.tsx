import React from 'react';
import { render } from '../../../jest/renderUtil';
import { CardHeader } from '../../../src/themes/card/CardHeader';

const setup = () => render(<CardHeader title="Test title" />);
describe('src >> themes >> card', () => {
  test('Should match snapshot', () => {
    const { toJSON } = setup();
    expect(toJSON()).toMatchSnapshot();
  });

  test('Should find title correctly', () => {
    const { getByText } = setup();
    expect(getByText('Test title')).toBeDefined();
  });
});
