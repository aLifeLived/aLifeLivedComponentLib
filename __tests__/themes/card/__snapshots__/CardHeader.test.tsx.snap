// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src >> themes >> card Should match snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "flex-end",
        "display": "flex",
        "paddingRight": 16,
        "paddingTop": 8,
      },
      Array [
        Object {},
      ],
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#282828",
          "display": "flex",
          "fontFamily": "SFUIText-Regular",
          "fontSize": 20,
          "lineHeight": 24,
        },
      ]
    }
  >
    Test title
  </Text>
</View>
`;
